artichoke-backend/src/extn/core/regexp/regexp_test.rb

```diff
@@ -29,4 +46,17 @@ def regexp_initialize_already_init_compiled
   end
 end
 
+# Since Ruby 3.0, Regexp literals are frozen by default.
+# https://github.com/ruby/ruby/pull/2705
+# https://github.com/ruby/ruby/pull/3676
+# Non-literal Regexp objects are still unfrozen by default.
+def regexp_initialize_only_literals_frozen_by_default
+  raise unless /abc/.frozen?
+
+  s = 'abc'
+  raise unless /#{s}/.frozen?
```

Comment:
I'm not sure this will pass due to some limitations of the mruby parser. If it fails, please comment this out and add a FIXME comment.